network/network_common.go

```diff
@@ -15,9 +15,60 @@ func (self *Network) Name() string {
 
 func (self *Network) Collect() (result interface{}, err error) {
 	result, err = getNetworkInfo()
+	interfaces := getMultiNetworkInfo()
+	if interfaces != nil {
+		interfaceMap, ok := result.(map[string]interface{})
+		if !ok {
+			return
+		}
+		interfaceMap["interfaces"] = interfaces
+	}
 	return
 }
 
+func getMultiNetworkInfo() (multiNetworkInfo []map[string]interface{}) {
+	ifaces, err := net.Interfaces()
+
+	if err != nil {
+		return nil
+	}
+	for _, iface := range ifaces {
+		_iface := make(map[string]interface{})
+		if iface.Flags&net.FlagUp == 0 || iface.Flags&net.FlagLoopback != 0 {
+			// interface down or loopback interface
+			continue
+		}
+		addrs, err := iface.Addrs()
+		if err != nil {
+			return nil
```

Comment:
Ditto. That's probably the wanted behavior but I prefer to ask.

Reply:
should've been `continue`. will amend

Reply:
https://github.com/DataDog/gohai/commit/240ea573b7a162111cc4f6392f96a3ca785a0f9b